jwst/pipeline/calwebb_spec3.py

```diff
@@ -1,230 +1,233 @@
+        """
+        self.log.info('Starting calwebb_spec3 ...')
+        asn_exptypes = ['science', 'background']
+
+        # Setup sub-step defaults
+        self.master_background.suffix = 'mbsub'
+        self.mrs_imatch.suffix = 'mrs_imatch'
+        self.outlier_detection.suffix = 'crf'
+        self.outlier_detection.save_results = self.save_results
+        self.resample_spec.suffix = 's2d'
+        self.resample_spec.save_results = self.save_results
+        self.cube_build.suffix = 's3d'
+        self.cube_build.save_results = self.save_results
+        self.extract_1d.suffix = 'x1d'
+        self.extract_1d.save_results = self.save_results
+        self.combine_1d.suffix = 'c1d'
+        self.combine_1d.save_results = self.save_results
+
+        # Retrieve the inputs:
+        # could either be done via LoadAsAssociation and then manually
+        # load input members into models and ModelContainer, or just
+        # do a direct open of all members in ASN file, e.g.
+        input_models = datamodels.open(input, asn_exptypes=asn_exptypes)
+
+        # Immediately update the ASNTABLE keyword value in all inputs,
+        # so that all outputs get the new value
+        for model in input_models:
+            model.meta.asn.table_name = op.basename(input_models.meta.table_name)
+
+        # For the first round of development we will assume that the input
+        # is ALWAYS an ASN. There's no use case for anyone ever running a
+        # single exposure through.
+
+        # Once data are loaded, store a few things for future use;
+        # some of this is here only for the purpose of creating fake
+        # products until the individual tasks work and do it themselves
+        exptype = input_models[0].meta.exposure.type
+        model_type = input_models[0].meta.model_type
+        output_file = input_models.meta.asn_table.products[0].name
+        self.output_file = output_file
+
+        # Find all the member types in the product
+        members_by_type = defaultdict(list)
+        product = input_models.meta.asn_table.products[0].instance
+        for member in product['members']:
+            members_by_type[member['exptype'].lower()].append(member['expname'])
+
+        if is_moving_target(input_models):
+            self.log.info("Assigning WCS to a Moving Target exposure.")
+            input_models = self.assign_mtwcs(input_models)
+
+        # If background data are present, call the master background step
+        if members_by_type['background']:
+            source_models = self.master_background(input_models)
+            source_models.meta.asn_table = input_models.meta.asn_table
+
+            # If the step is skipped, do the container splitting that
+            # would've been done in master_background
+            if self.master_background.skip:
+                source_models, bkg_models = split_container(input_models)
+                # we don't need the background members
+                bkg_models.close()
```

Comment:
Changes here.  Sorry for the big diff. For some reason this file had been committed to the repo with Windows line endings ^M.  This converts it back to unix type, as for all the other files in the repo.

Reply:
Windows?! Arrrrrrrgh - make it stop!